spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/reactive/ReactorLoadBalancerClientAutoConfiguration.java

```diff
@@ -61,16 +63,18 @@ public ReactorLoadBalancerExchangeFilterFunction loadBalancerExchangeFilterFunct
 	public RetryableLoadBalancerExchangeFilterFunction retryableLoadBalancerExchangeFilterFunction(
 			ReactiveLoadBalancer.Factory<ServiceInstance> loadBalancerFactory, LoadBalancerProperties properties,
 			LoadBalancerRetryPolicy retryPolicy,
-			ObjectProvider<List<LoadBalancerClientRequestTransformer>> transformers) {
+			ObjectProvider<List<LoadBalancerClientRequestTransformer>> transformers,
+			LoadBalancerPropertiesFactory propertiesFactory) {
 		return new RetryableLoadBalancerExchangeFilterFunction(retryPolicy, loadBalancerFactory, properties,
-				transformers.getIfAvailable(Collections::emptyList));
+				transformers.getIfAvailable(Collections::emptyList), propertiesFactory);
 	}
 
 	@ConditionalOnMissingBean
 	@ConditionalOnProperty(value = "spring.cloud.loadbalancer.retry.enabled", havingValue = "true")
```

Comment:
@OlgaMaciaszek @spencergibb 

I see an issue with enabling`LoadBalancerRetryPolicy` by `spring.cloud.loadbalancer.retry.enabled `.

I think we can use `spring.cloud.loadbalancer.retry.enabled` property for enabling retry in general. And then we should check in `policy` (and in other beans) that retry is enabled for a specific service (it has not implemented yet).

Reply:
Yes, we definitely should keep `spring.cloud.loadbalancer.retry.enabled` to be used as a global switch, but we could probably also be able to set `spring.cloud.loadbalancer.client.clientName.retry.enabled` per client - wdyt?

Reply:
But I'm not sure how to implement switching by `spring.cloud.loadbalancer.client.clientName.retry.enabled` for reactive loadBalancer. Reactive loadBalance has two functions - with (`RetryableLoadBalancerExchangeFilterFunction`) and without (`ReactorLoadBalancerExchangeFilterFunction`) retry. and as I see there no possibility to create pipeline for a few filters (we can but we don't have an order guaranty). so I tried to make it in retry filter - [RetryableLoadBalancerExchangeFilterFunction.java](https://github.com/spring-cloud/spring-cloud-commons/pull/947/files#diff-a8e7d22f94822da32e047a7bac0fd8c6cc36fdffb667bd66682121a06b2e636fR128)

Reply:
That looks fine, but I'm thinking we also need a way to handle global properties; I imagine we would do so with retries, but also possibly with caching (we should allow disabling it globally - could you add this?) and possibly other functionalities in the future, so apart from the method `getLoadBalancerProperties(String serviceName)` in the factory, we could also have sth like `getGlobalLoadBalancerProperties()` (that would just return the non-`client`-prefixed properties - then for the major we would have to move most of the properties to the `spring.cloud.loadbalancer.client.clientName` - prefixed, and only leave the global ones prefixed with just `spring.cloud.loadbalancer`)  and in cases where we want to offer a global off switch (or some other global property) we could verify first the global properties. wdyt?

Reply:
I'll add `getGlobalLoadBalancerProperties()` method to the factory. But I don't see where we can use it.

For caching we use a different property class `LoadBalancerCacheProperties.java`. Change cache classes to use the new way of configuration not enough, functionality also should be changed (`DefaultLoadBalancerCacheManager` and `CaffeineBasedLoadBalancerCacheManager`). I think it should be done in a separate thread.

Reply:
Another option would be to rework the way the `ServiceInstanceListSupplier` beans are being created and, in place of using `@ConditionalOnProperty` during bean creation, use the properties bean/ factory passed in constructor and then create the bean, for example, `new RetryAwareServiceInstanceListSupplier(delegate)` (see [usage from current code](https://github.com/spring-cloud/spring-cloud-commons/blob/6ed50fcbc7ab9e41eac5eb4641cde952fa81823e/spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java#L197-L199)) if the flag from properties is `true` and just return the delegate bean if it's `false`. Then we would be able to always give the highest priority for `spring.cloud.loadbalancer.client.xxx.property` and fall back to `spring.cloud.loadbalancer.property` consistently for all the features. @Ferioney @spencergibb please let me know what you think about the various possible approaches/ propose others.